scrapinghub/client.py

```diff
@@ -1051,6 +1051,122 @@ def post(self, _value, **kwargs):
+
+    Usage:
+
+    - get all frontiers from a project::
+        >>> project.frontiers.iter()
+        <list_iterator at 0x103c93630>
+
+    - list all frontiers
+        >>> project.frontiers.list()
+        ['test', 'test1', 'test2']
+
+    - get a frontier by name
+        >>> project.frontiers.get('test')
+        <scrapinghub.client.Frontier at 0x1048ae4a8>
+
+    - flush data of all frontiers of a project
+        >>> project.frontiers.flush()
+
+    - close batch writers of all frontiers of a project
+        >>> project.frontiers.close()
+    """
+    def __init__(self, *args, **kwargs):
+        super(Frontiers, self).__init__(*args, **kwargs)
+        self._proxy_methods(['close', 'flush'])
+
+    def get(self, name):
+        return Frontier(self._client, self, name)
+
+    def iter(self):
+        return iter(self.list())
+
+    def list(self):
+        return next(self._origin.apiget('list'))
+
+    @property
+    def newcount(self):
+        return self._origin.newcount
+
+
+class Frontier(object):
+    """Representation of a frontier object.
+
+    Not a public constructor: use :class:`Frontiers` instance to get a
+    :class:`Frontier` instance. See :meth:`Frontiers.get` method.
+
+    Usage:
+
+    - get iterator with all slots
+        >>> frontier.iter_slots()
+        <list_iterator at 0x1030736d8>
+
+    - list all slots
+        >>> frontier.list_slots()
+        ['example.com', 'example.com2']
+
+    - add request to a slot
+        >>> data = [{'fp': 'page1.html', 'p': 1, 'qdata': {'depth': 1}}]
+        >>> frontier.add('example.com', data)
+
+    - read requests from a slot
+        >>> data = frontier.read('example.com')
+        >>> data
+        <generator object jldecode at 0x1049aa9e8>
+        >>> list(data)
+        [{'id': '0115a8579633600006',
+          'requests': [['page1.html', {'depth': 1}]]}]
+
+    - delete a request from a slot
+        >>> frontier.delete('example.com', '0115a8579633600006')
+
+    - flush frontier data
+        >>> frontier.flush()
+
+    - flush data for a given slot only
+        >>> frontier.flush('example.com')
+
+    - delete slot by name
+        >>> frontier.delete_slot('example.com')
+    """
+
+    def __init__(self, client, frontiers, name):
+        self.key = name
+        self._client = client
+        self._frontiers = frontiers
+
+    def add(self, slot, fps):
+        return self._frontiers._origin.add(self.key, slot, fps)
+
+    def read(self, slot, mincount=None):
+        return self._frontiers._origin.read(self.key, slot, mincount)
+
+    def delete(self, slot, ids):
+        return self._frontiers._origin.delete(self.key, slot, ids)
+
+    def delete_slot(self, slot):
+        return self._frontiers._origin.delete_slot(self.key, slot)
+
+    def iter_slots(self):
+        return iter(self.list_slots())
+
+    def list_slots(self):
+        return next(self._frontiers._origin.apiget((self.key, 'list')))
```

Comment:
Just a proposal - maybe we should add another entity - Slot or FrontierSlot? Frontier.iter() will go through available slot names, Frontier.get() will get Slot object, Slot.iter() or Slot.list() will do what read() does now, Slot.delete(ids) will delete data from the slot, Frontier.delete(name) will delete whole slot